### PR TITLE
Internal secrets management from library

### PR DESCRIPTION
**Pull request by juditnovak**
`data_platform_libs/data_interface v26`, together with additional tests.

The library now includes support for handling Peer Relation Data (similar to Cross-charm Relation Data), hiding all complexity of choice between databag and Juju Secrets.

The same code is also tested against Juju 3.1.7: https://github.com/canonical/postgresql-operator/pull/314

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -102,6 +103,30 @@ def __init__(self, *args):
...
+            relation_name=PEER,
+            additional_secret_fields=[
+                "monitoring-password",
+                "operator-password",
+                "replication-password",
+                "rewind-password",
+            ],
+            secret_field_name=SECRET_INTERNAL_LABEL,
+            deleted_label=SECRET_DELETED_LABEL,
+        )
```
Comment (marceloneppel) [inline src/charm.py:129]: Is having those secrets (password) in both `peer_relation_app` and `peer_relation_unit` necessary?

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -1570,3 +1445,266 @@ def test_client_relations(self):
...
+            assert (
+                "Non-existing secret 'non-existing-secret' was attempted to be removed"
+                in self._caplog.text
+            )
+
+    @parameterized.expand([("app", True), ("unit", True), ("unit", False)])
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.PostgresqlOperatorCharm._on_leader_elected")
+    @patch("charm.JujuVersion.has_secrets", new_callable=PropertyMock, return_value=True)
+    def test_migartion_from_databag(self, scope, is_leader, _, __):
```
Comment (marceloneppel) [inline tests/unit/test_charm.py:1668]:
```suggestion
    def test_migration_from_databag(self, scope, is_leader, _, __):
```

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -1570,3 +1445,266 @@ def test_client_relations(self):
...
+        assert self.harness.charm.get_secret(scope, "operator-password") == "blablabla"
+        assert "operator-password" not in self.harness.get_relation_data(
+            self.rel_id, getattr(self.charm, scope).name
+        )
+
+    @parameterized.expand([("app", True), ("unit", True), ("unit", False)])
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.PostgresqlOperatorCharm._on_leader_elected")
+    @patch("charm.JujuVersion.has_secrets", new_callable=PropertyMock, return_value=True)
+    def test_migartion_from_single_secret(self, scope, is_leader, _, __):
```
Comment (marceloneppel) [inline tests/unit/test_charm.py:1690]:
```suggestion
    def test_migration_from_single_secret(self, scope, is_leader, _, __):
```

**Review by marceloneppel — COMMENTED** [review]
Nice tests!

File: src/charm.py (hunk 1 above)
Comment (juditnovak) [reply src/charm.py:129]: No, not at all :sweat_smile: 
I was just cutting corners in the POC code :sweat_smile: 

(I'm leaving the conversation up until I'd fix the issue.)

**Review by dragomirp — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Review by marceloneppel — APPROVED** [review]

**Pull request merged**